src/transformers/models/vocos/feature_extraction_vocos.py

```diff
@@ -0,0 +1,237 @@
+    Fourier Transform (STFT).
+
+    Args:
+        feature_size (`int`, *optional*, defaults to 100):
+            The feature dimension of the extracted features.
+        sampling_rate (`int`, *optional*, defaults to 24000):
+            The sampling rate at which the audio files should be digitalized expressed in hertz (Hz).
+        num_mel_bins (`int`, *optional*, defaults to 100):
+            Number of Mel-frequency bins.
+        n_fft (`int`, *optional*, defaults to 1024):
+            Size of the Fourier transform.
+        hop_length (`int`, *optional*, defaults to 256):
+            Length of the overlapping windows for the STFT used to obtain the Mel Frequency coefficients.
+        padding (`str`, *optional*, defaults to `"center"`):
+            Symmetric padding if 'same' and center padding if 'center'.
+        padding_value (`float`, *optional*, defaults to 0.0):
+            Padding value used to pad the audio. Should correspond to silences.
+        return_attention_mask (`bool`, *optional*, defaults to `True`):
+            Whether to return the attention mask. If left to the default, it will return the attention mask.
+
+            [What are attention masks?](../glossary#attention-mask)
+
+    """
+
+    model_input_names = ["audio_spectrogram"]
+
+    def __init__(
+        self,
+        feature_size=100,
+        sampling_rate=24000,
+        num_mel_bins=100,
+        n_fft=1024,
+        hop_length=256,
+        padding="center",
+        padding_value=0.0,
+        return_attention_mask=True,
+        **kwargs,
+    ):
+        super().__init__(
+            feature_size=feature_size,
+            sampling_rate=sampling_rate,
+            padding_value=padding_value,
+            return_attention_mask=return_attention_mask,
+            **kwargs,
+        )
+
+        self.n_fft = n_fft
+        self.hop_length = hop_length
+        self.win_length = n_fft
+        self.num_mel_bins = num_mel_bins
+        if padding not in ["center", "same"]:
+            raise ValueError("Padding must be either `center` or `same`.")
+        self.padding = padding
+
+        requires_backends(self, ["torchaudio"])
+        self.mel_filters = torchaudio.transforms.MelSpectrogram(
+            sample_rate=self.sampling_rate,
+            n_mels=self.num_mel_bins,
+            n_fft=self.n_fft,
+            hop_length=self.hop_length,
+            center=(self.padding == "center"),
+            power=1,
+            window_fn=torch.hann_window,
+        )
+
+    def __call__(
+        self,
+        audio: AudioInput,
+        padding: Optional[Union[bool, str, PaddingStrategy]] = True,
+        max_length: Optional[int] = None,
+        truncation: Optional[bool] = False,
+        sampling_rate: Optional[int] = None,
+        return_tensors: Optional[Union[str, TensorType]] = None,
+        return_audio_only: Optional[bool] = False,
```

Comment:
for encodec variant, `return_audio_only` is used to cut processing before feature extraction (essentially to use input preperation and padding from the feature extractor)